src/modules/advanced-custom-attributes/index.js

```diff
@@ -0,0 +1,191 @@
+					const match = _match.substr( 1, _match.length - 2 )
+					if (
+						! INVALID_BLOCK_ATTRIBUTES.includes( match ) &&
+						blockProps.attributes.hasOwnProperty( match ) &&
+						! isUndefined( blockProps.attributes[ match ] )
+					) {
+						value = value.replace( _match, striptags( blockProps.attributes[ match ].toString() ) )
+					}
+				} )
+			}
+			customAttributes[ key ] = value
+		} )
+	} catch {}
+
+	return {
+		...extraProps,
+		...omit( customAttributes, INVALID_HTML_ATTRIBUTES ),
+	}
+}
+
+const addAttributes = attributes => ( {
+	...attributes,
+	customAttributes: {
+		type: 'array',
+		default: [],
+	},
+} )
+
+export const CustomAttributesControl = props => {
+	const [ customAttributes, setCustomAttributes ] = useState(
+		Array.isArray( props.value ) ?
+			props.value.map( attribute => {
+				const [ key, _value ] = attribute
+				const value = `"${ _value }"`
+				return [ key, value ].join( '=' )
+			} ).join( ' ' ) :
+			''
+	)
+
+	const [ error, setError ] = useState( '' )
+
+	useEffect( () => {
+		const timeout = setTimeout( () => {
+			try {
+				if ( customAttributes.trim() === '' ) {
+					props.onChange( [] )
+					setError( '' )
+					return
+				}
+
+				let tempCustomAttributes = customAttributes
+				const customAttributesMatch = customAttributes.match( /(.*?(?=\=))\="[^\"]*"/g )
+				if ( customAttributes.trim() !== '' && ! customAttributesMatch ) {
+					throw new Error( __( 'Not a valid attribute. Input must be of pattern key1="value1" key2="value2".', i18n ) )
+				}
+
+				// Trim the original input value to check if there are unnecessary characters added.
+				customAttributesMatch.forEach( attribute => {
+					tempCustomAttributes = tempCustomAttributes.replace( attribute.trim(), '' )
+				} )
+
+				if ( tempCustomAttributes.trim() !== '' ) {
+					throw new Error( sprintf( __( "Unexpected syntax: '%s'.", i18n ), tempCustomAttributes.trim() ) )
+				}
+
+				const newCustomAttributes = []
+				customAttributesMatch.forEach( attribute => {
+					const [ key, ..._value ] = attribute.trim().split( '=' )
+					const value = JSON.parse( _value.join( '=' ) )
+
+					/**
+					 * Checks if the attribute key and value can be a valid html attribute.
+					 * Throws an error if not valid.
+					 */
+					document.createElement( 'div' ).setAttribute( key, value )
+
+					/**
+					 * Checks if the attribute key and value can be a valid react prop
+					 * Throws an error if not valid.
+					 */
+					try {
+						const dummyNode = document.createElement( 'div' )
+						render( <div { ...{ [ key ]: value } } />, dummyNode )
+						unmountComponentAtNode( dummyNode )
+
+						if ( INVALID_HTML_ATTRIBUTES.includes( key ) ) {
+							throw new Error()
+						}
+					} catch {
+						throw new Error( sprintf( __( "Attribute key '%s' is not allowed.", i18n ), key ) )
+					}
+
+					newCustomAttributes.push( [ key, value ] )
+				} )
+
+				props.onChange( newCustomAttributes )
+				setError( '' )
```

Comment:
I'd like to recommend something simpler. Please see https://developer.mozilla.org/en-US/docs/Web/API/Element/attributes

You can grab the input of the user, then pass it on as `innerHTML` of a dummy element:

```js
const el = document.createElement( 'div' )
el.innerHTML = `<div ${ stringValue } />`
// Get all the "fixed" attributes:
const attrNodeMap = el.children[0].attributes // This is a NamedNodeMap
const attributes = {}
for (let i = attrNodeMap.length - 1; i >= 0; i--) {
    attributes[ attrNodeMap[ i ].name ] = attrNodeMap[ i ].value
}
```

What you can do is instead of detecting and displaying an error, you can have the above method adjust the value when the input is blurred